statsmodels/base/_penalized.py

```diff
@@ -0,0 +1,194 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Sun May 10 08:23:48 2015
+
+Author: Josef Perktold
+License: BSD-3
+"""
+
+import numpy as np
+from ._penalties import NonePenalty
+from statsmodels.tools.numdiff import approx_fprime_cs, approx_fprime
+
+class PenalizedMixin(object):
+    """Mixin class for Maximum Penalized Likelihood
+
+    Parameters
+    ----------
+    args and kwds for the model super class
+    penal : None or instance of Penalized function class
+        If penal is None, then NonePenalty is used.
+    pen_weight : float or None
+        factor for weighting the penalization term.
+        If None, then pen_weight is set to nobs.
+
+
+    TODO: missing **kwds or explicit keywords
+
+    TODO: do we adjust the inherited docstrings?
+    We would need templating to add the penalization parameters
+
+    """
+
+    def __init__(self, *args, **kwds):
+
+        # pop extra kwds before calling super
+        self.penal = kwds.pop('penal', None)
+        self.pen_weight =  kwds.pop('pen_weight', None)
+
+        super(PenalizedMixin, self).__init__(*args, **kwds)
```

Comment:
does this do anything here?  base class is `object`.  Pretty harmless, except that having `args` prevents making `kwds` explicit, and the `kwds.pop` pattern is a recipe for silently ignoring misspelled kwargs.

Reply:
This is important. With multiple inheritance the `super` is the model class in this case, i.e. GLM or Poisson in the examples, and then all the way up to base.model.